src/code/containers/challenge-container-selector.js

```diff
@@ -66,15 +75,14 @@ class ChallengeContainerSelector extends Component {
 function mapStateToProps (state) {
   return {
     authoring: state.authoring,
-    case: state.case,
-    challenge: state.challenge
+    currentRouteSpec: state.routeSpec
```

Comment:
I did this intentionally to differentiate the `routeSpec` from the `routeParams`, since I kept confusing them. I only changed this one because I didn't want to touch the magically passed `routeParams`.